src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/NovelAdjacencyReferenceLocations.java

```diff
@@ -77,50 +123,99 @@ protected NovelAdjacencyReferenceLocations(final Kryo kryo, final Input input) {
+                    leftBreakpointCoord  = leftReferenceInterval.getEnd() - homologyLen;
+                    rightBreakpointCoord = rightReferenceInterval.getStart() - 1;
+                } else if (ca.strandSwitch == StrandSwitch.FORWARD_TO_REVERSE){
+                    leftBreakpointCoord  = leftReferenceInterval.getEnd() - homologyLen;
+                    rightBreakpointCoord = rightReferenceInterval.getEnd();
                 } else {
-                    leftBreakpointCoord = leftReferenceInterval.getEnd() - homologyLen;
+                    leftBreakpointCoord  = leftReferenceInterval.getStart() - 1;
+                    rightBreakpointCoord = rightReferenceInterval.getStart() + homologyLen - 1;
                 }
-                rightBreakpointCoord = rightReferenceInterval.getStart() - 1;
-            } else {
-                leftBreakpointRefContig = rightBreakpointRefContig = ca.regionWithLowerCoordOnContig.referenceSpan.getContig();
-                leftBreakpointCoord     = complication.getDupSeqRepeatUnitRefSpan().getStart() - 1;
-                rightBreakpointCoord    = complication.getDupSeqRepeatUnitRefSpan().getEnd();
             }
-        } else { // inversion and simple deletion & insertion
-            final SimpleInterval leftReferenceInterval, rightReferenceInterval;
-            if (ca.isForwardStrandRepresentation) {
-                leftReferenceInterval  = ca.regionWithLowerCoordOnContig.referenceSpan;
-                rightReferenceInterval = ca.regionWithHigherCoordOnContig.referenceSpan;
-            } else {
-                leftReferenceInterval  = ca.regionWithHigherCoordOnContig.referenceSpan;
-                rightReferenceInterval = ca.regionWithLowerCoordOnContig.referenceSpan;
-            }
-            leftBreakpointRefContig  = leftReferenceInterval.getContig();
-            rightBreakpointRefContig = rightReferenceInterval.getContig();
+        } else {
             if (ca.strandSwitch == StrandSwitch.NO_SWITCH) {
-                leftBreakpointCoord  = leftReferenceInterval.getEnd() - homologyLen;
-                rightBreakpointCoord = rightReferenceInterval.getStart() - 1;
-            } else if (ca.strandSwitch == StrandSwitch.FORWARD_TO_REVERSE){
-                leftBreakpointCoord  = leftReferenceInterval.getEnd() - homologyLen;
-                rightBreakpointCoord = rightReferenceInterval.getEnd();
+                if (ca.isForwardStrandRepresentation) { // there's no absolute difference between left and right here because they map to different chromosome, left/right could be later defined by dictionary when outputting VCF
+                    if (0 > IntervalUtils.compareContigs(ca.regionWithLowerCoordOnContig.referenceSpan,
+                            ca.regionWithHigherCoordOnContig.referenceSpan, referenceDictionary)) {
+                        leftBreakpointRefContig  = ca.regionWithLowerCoordOnContig.referenceSpan.getContig();
```

Comment:
Again, having so many similar but subtly different sets of lines that set `leftBreakpointCoord`, etc. is quite difficult to read. Here's another suggestion for refactoring this code -- could you decide on a "default" ordering of the two CA regions, and then apply reversals or homology adjustments as operations on your coordinate set? Then it would look like:

coords = default coordinates
if (strand switch FR) reverse coords
etc..

That may or may not work in conjunction with my suggestion of refactoring to subclasses that I talked about above, so feel free to take one suggestion or the other.

Reply:
I refactored this 100+ line of branching into a class hierarchy in this new commit. Some refactoring still needs to be done for `ChimericAlignment`, but it is more suitable for the next planned refactoring PR.